trace/src/main/org/jetbrains/lincheck/trace/TracePostprocessor.kt

```diff
@@ -0,0 +1,142 @@
+     * ```
+     *
+     */
+    private fun TRMethodCallTracePoint.compressDefaultPairs(reader: LazyTraceReader): TRMethodCallTracePoint {
+        if (events.size != 1) return this
+        val singleChild = loadChild(reader, tracePoint = this, 0)
+        if (singleChild == null) return this
+
+        if (singleChild !is TRMethodCallTracePoint) return this
+        if (!haveSameOwner(className, singleChild.className) || !isDefaultPair(methodName, singleChild.methodName)) return this
+
+        return combineCallNodes(this, singleChild)
+    }
+
+    /**
+     * Compresses `.access$` calls.
+     *
+     * The `.access$` methods are generated by the Kotlin compiler to access otherwise inaccessible members
+     * (e.g., private) from lambdas, inner classes, etc.
+     *
+     * For example:
+     *
+     * ```
+     * A.access$callMe() at A.operation(A.kt:N)
+     *  A.callMe() at A.access$callMe(A.kt:N)
+     * ```
+     *
+     * will be collapsed into:
+     *
+     * ```
+     * A.callMe() at A.operation(A.kt:N)
+     * ```
+     *
+     */
+    private fun TRMethodCallTracePoint.compressAccessPairs(reader: LazyTraceReader): TRMethodCallTracePoint {
+        if (events.size != 1) return this
+        val singleChild = loadChild(reader, tracePoint = this, 0)
+        if (singleChild == null) return this
+
+        if (singleChild !is TRMethodCallTracePoint) return this
+        if (!haveSameOwner(className, singleChild.className) || !isAccessPair(methodName, singleChild.methodName)) return this
+
+        return combineCallNodes(this, singleChild)
+    }
+
+    /**
+     * Compresses synthetic field access methods (`access$get` and `access$set`).
+     *
+     * These methods are generated by the Kotlin compiler when a lambda or inner class accesses a private field.
+     * This function removes the synthetic access method and keeps only the actual field access event.
+     *
+     * For example:
+     *
+     * ```
+     * A.access$get_field(A#1) at A.operation(A.kt:N)
+     *   field ➜ value at A.access$get_field(A.kt:N)
+     * ```
+     *
+     * will be collapsed into:
+     *
+     * ```
+     * field ➜ value at A.operation(A.kt:N)
+     * ```
+     *
+     * This is different from `fun$access`, which is addressed in [compressAccessPairs].
+     *
+     */
+    private fun TRMethodCallTracePoint.compressSyntheticFieldAccess(reader: LazyTraceReader): TRTracePoint {
+        if (events.size != 2 && events.size != 3) return this
+        if (!isSyntheticFieldAccess(methodName)) return this
+
+        // get and set method versions have different number of children (2 and 3, correspondingly)
+        // the correct access node is stored at the last position
+        val actualNode = if (isSyntheticGetFieldAccess(methodName)) loadChild(reader, tracePoint = this, 1) ?: return this
+                         else loadChild(reader, tracePoint = this, 2) ?: return this
+
+        actualNode.replaceCodeLocationId(codeLocationId)
```

Comment:
Why do you need to change `codeLocationId` ?

Reply:
IN order for clicking in the trace sidebar to work and point to the correct code location

Reply:
Please add a comment then.